nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -645,8 +645,85 @@ impl StringPrototype {
         ))
     }
 
-    fn replace(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    /// ### [22.1.3.19 String.prototype.replace ( searchValue, replaceValue )](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.replace)
+    fn replace(agent: &mut Agent, this_value: Value, args: ArgumentsList) -> JsResult<Value> {
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        let o = require_object_coercible(agent, this_value)?;
+
+        let search_value = args.get(0);
+        let replace_value = args.get(1);
+
+        // 2. If searchValue is neither undefined nor null, then
+        if !search_value.is_null() && !search_value.is_undefined() {
+            let symbol = WellKnownSymbolIndexes::Replace.into();
+            // If replacer is not undefined, Return ? Call(replacer, searchValue, « O, replaceValue »).
+            if let Ok(Some(replacer)) = get_method(agent, search_value, symbol) {
+                return call_function(
+                    agent,
+                    replacer,
+                    search_value,
+                    Some(ArgumentsList(&[o, replace_value])),
+                );
+            }
+        }
+
+        // 3. Let s be ? ToString(O).
+        let s = to_string(agent, o)?;
```

Comment:
**suggestion:** Here we could do an "expected" early path:

```rs
if let (Ok(s), Ok(search_string), Ok(replace_string)) = (String::try_from(o), String::try_from(search_value), String::try_from(replace_value)) { ... } else { ... }
```
The expected path this function gets called is something like `"asdbarbazfoobot".replace("bar", "foo")`. So, we can skip the is_callable etc. tests if we find ourselves on this "normal" path.